tests/SerdeTestCases.php

```diff
@@ -1187,6 +1190,50 @@ public function datetime_fields_support_custom_output_format(): void
 
     }
 
+    #[Test]
+    public function unixtime_fields_support(): void
+    {
+        $s = new SerdeCommon(formatters: $this->formatters);
+
+        $timeString = '@1656958942.123456';
+
+        $stamp = new \DateTime($timeString);
+        $stampImmutable = new \DateTimeImmutable($timeString);
+        $data = new UnixTimeExample(
+            seconds: $stamp,
+            milliseconds: $stampImmutable,
+            microseconds: $stampImmutable,
+            nanoseconds: $stampImmutable
+        );
+
+        $serialized = $s->serialize($data, $this->format);
+
+        $this->unixtime_validate($serialized);
+
+        // Because some of the exported formats involve data loss,
+        // we don't actually expect the exact same thing back.
+        $expected = new UnixTimeExample(
+            seconds: new \DateTime('@1656958942'),
+            milliseconds: new \DateTimeImmutable('@1656958942.123'),
+            microseconds: new \DateTimeImmutable('@1656958942.123456'),
+            nanoseconds: new \DateTimeImmutable('@1656958942.123456')
+        );
+
+        $result = $s->deserialize($serialized, from: $this->format, to: $data::class);
+
+        self::assertEquals($expected, $result);
+
+    }
+
+    public function unixtime_validate(mixed $serialized): void {
```

Comment:
Not quite; this method should be empty in this class, and the implementation of it moved down to `ArrayBasedFormatterTest`.  There's already an arrayify there you can use.

Stream-based formatters cannot easily use an "arrayify" approach, so they have the flexibility to do something else.

Reply:
I'm on a vacation at the moment, but should have some time in the near future to take care of this. At the latest, I'll get to it next week.